### DISCUSSION TITLE
[Feature Request]: setFirstOption only for Multiselect filter

### DISCUSSION THREAD

**Discussion by trippo** [rappasoft/laravel-livewire-tables#1497] [Feature Requests]
### What happened?

`setFirstOption` method is defined only for Multi-select Filters

But documentation says that it is defined also for Select Filter https://rappasoft.com/docs/laravel-livewire-tables/v2/filters/creating-filters#content-option-groups

### How to reproduce the bug

```
SelectFilter::make('Active')
            ->options([
                '' => 'All',
                'Open' => [
                    1 => 'Type A',
                    2 => 'Type B',
                    3 => 'Type C',
                ],
                'Closed' => [
                    24 => 'Type X',
                    25 => 'Type Y',
                    26 => 'Type Z',
                ],
            ])
            ->setFirstOption('All Tags'),
```

### PHP Version

None

### Theme

None

### Error Message

`Call to undefined method Rappasoft\LaravelLivewireTables\Views\Filters\SelectFilter::setFirstOption()`

**Comment by lrljoe** [comment]
Thanks for raising, I'm addressing this in V3 as we speak, but will backport to v2 once the v3 Beta is out.

You can of course pass a blank value at the start of the array in the meantime!

E.g.

```
  SelectFilter::make('E-mail Verified', 'email_has_been_verified')
      ->setFilterPillTitle('Has Verified E-Mail')
      ->options([
          '' => 'Any',
          'yes' => 'Yes',
          'no' => 'No',
      ])
      ->filter(function (Builder $builder, string $value) {
          if ($value === 'yes') {
              $builder->whereNotNull('users.email_verified_at');
          } elseif ($value === 'no') {
              $builder->whereNull('users.email_verified_at');
          }
      }),
```

Or in your case:
```
SelectFilter::make('Active')
            ->options([
                '' => 'All Tags',
                'Open' => [
                    1 => 'Type A',
                    2 => 'Type B',
                    3 => 'Type C',
                ],
                'Closed' => [
                    24 => 'Type X',
                    25 => 'Type Y',
                    26 => 'Type Z',
                ],
            ]),
```
Note that you should definitely be defining a filter() callback, otherwise you may experience a one lifecycle delay in v2!

**Comment by lrljoe** [comment]
This is fixed for SelectFilter in the V3 Beta.

I will backport this to v2

**Comment by lrljoe** [comment]
Needs backporting still

**Comment by lrljoe** [comment]
Just updating, as it's technically a feature request.

I've merged v3 development into v3-master now, so it's very close to actual release.

Will be backporting this into v2 for all select options once v3 is actually out

**Comment by lrljoe** [comment]
Noting that with your approach, you don't actually need setFirstOption, that's more for if you're not merging arrays how you are.

This is still in the backport queue tho!
